File: 0605-can-place-flowers/0605-can-place-flowers.java
class Solution {
    public int plant(int[] flowerbed, int n, int c, int prev, int cur){
        int flower = cur;
        if (c == flowerbed.length-1 || n == 0) {//if you are at the end of the array
            if ((prev == -1 || prev == 0) && flower == 0){
                n -= 1;
            }
            return n;
        }
        if (cur == 0 && flowerbed[c+1] == 0 && (prev == -1 || prev == 0)){
            return plant(flowerbed, n-1, ++c, 1, flowerbed[c]);
        } else {
            return plant(flowerbed, n, ++c, flower, flowerbed[c]);
        }
    }

    public boolean placer(int[] flowerbed, int n){
        if (n == 0) return true;
        int result = plant(flowerbed, n, 0, -1, flowerbed[0]);
        if (result != 0) return false;
        return true; 
    }
    public boolean canPlaceFlowers(int[] flowerbed, int n) {
        return placer(flowerbed, n);
    }

}

File: 0605-can-place-flowers/0605-can-place-flowers.cs
public class Solution {
    public bool CanPlaceFlowers(int[] flowerbed, int n) {
        
        if (flowerbed.Length == 1){
            if (flowerbed[0] == 0 && n <= 1) return true;
            else if (flowerbed[0] == 1 && n == 0) return true;
            else return false;
        }
        int prev = -1;
        for (int i = 0; i < flowerbed.Length-1; i ++){
            if (n == 0) return true;



            // if between two zero
            if (flowerbed[i] == 0 && prev == 0 && flowerbed[i+1] == 0){
                prev = 1;
                n -= 1;
            }else if (prev == -1 && flowerbed[i] == 0 && flowerbed[i+1] == 0) {
                prev = 1;
                n-=1;
            }
            else {
                prev = flowerbed[i];
            }
        }
        if (prev == 0 && flowerbed[flowerbed.Length-1] == 0) n-=1;


        return (n==0) ? true : false;
    }
}